x-pack/solutions/observability/plugins/apm_data_access/server/lib/helpers/create_es_client/create_apm_event_client/get_request_base.ts

```diff
@@ -39,13 +40,8 @@ export function getRequestBase(options: {
 
   const index = processorEventsToIndex(events, options.indices);
 
-  const filters: ESFilter[] = [
-    {
-      terms: {
-        [PROCESSOR_EVENT]: events,
-      },
-    },
-  ];
+  const filters: ESFilter[] =
+    options.skipProcessorEventFilter === true ? [] : [{ terms: { [PROCESSOR_EVENT]: events } }];
```

Comment:
Maybe we could do this on line 36, to prevent it from calling `getProcessorEventForDocumentType` unnecessarily. Wdyt?